implementations/code_df/commit.py

```diff
@@ -0,0 +1,89 @@
+import pandas as pd
+from .metric import Metric
+from . import utils
+
+
+class Commit(Metric):
+    """
+    Initilizes self.df, the dataframe with one commit per row.
+
+    :param items: A list of dictionaries.
+        Each element is a Perceval dictionary, obtained from a JSON
+        file or from Perceval directly.
+
+    :param date_range: A tuple which represents the period of interest
+        It is of the form (since, until), where since and until are strings.
+        Either, or both can be None. If, for example, since is None, that
+        would mean that all commits from the first commit to the commit
+        who last falls inside the until range will be included.
+
+    :param issrccode_class: A reference to IsSourceCode class.
+        It is used to determine what comprises source code.
+        """
+
+    def __init__(self, items, date_range=(None, None), issrccode_class=None):
+
+        self.issrccode_class = issrccode_class
+        (self.since, self.until) = date_range
+
+        clean_items = list()
+
+        for line in items:
+            commit = self._flatten_data(line)
+            if commit is not None:
+                clean_items.append(commit)
+
+        self.df = pd.DataFrame(clean_items)
+
+        if self.since is None:
+            self.since = utils.get_date(self.df, 'since')
+        if self.until is None:
+            self.until = utils.get_date(self.df, 'until')
```

Comment:
@jgbarah, though I moved everything to `_flatten_data`, I had to move the `since, until` population in case they are `None` back to `__init__`, since they are to be populated with the minimum value of all commits. Later, when we modify `compute_timeseries` (to avoid this problem), I'll remove this part. 
`_flatten_data` works for each commit, and hence it does not seem proper to keep a track of min/max date among all commits and at the last iteration, assign that to `since` or `until`.